src/data_types/w3c/credential.rs

```diff
@@ -0,0 +1,427 @@
+                "CredentialAttributes validation failed: empty list has been passed".into(),
+            );
+        }
+        Ok(())
+    }
+}
+
+impl From<&CredentialValues> for CredentialAttributes {
+    fn from(values: &CredentialValues) -> Self {
+        CredentialAttributes(
+            values
+                .0
+                .iter()
+                .map(|(attribute, values)| {
+                    (
+                        attribute.to_owned(),
+                        CredentialAttributeValue::Attribute(values.raw.to_owned()),
+                    )
+                })
+                .collect(),
+        )
+    }
+}
+
+impl CredentialAttributes {
+    pub(crate) fn add_attribute(&mut self, attribute: String, value: CredentialAttributeValue) {
+        self.0.insert(attribute, value);
+    }
+
+    pub(crate) fn add_predicate(
+        &mut self,
+        attribute: String,
+        predicate: PredicateAttribute,
+    ) -> Result<()> {
+        match self.0.get_mut(&attribute) {
+            Some(value) => match value {
+                CredentialAttributeValue::Attribute(_) => {
+                    return Err(err_msg!("Predicate cannot be added for revealed attribute"));
+                }
+                CredentialAttributeValue::Predicate(predicates) => predicates.push(predicate),
+            },
+            None => {
+                self.0.insert(
+                    attribute,
+                    CredentialAttributeValue::Predicate(vec![predicate]),
+                );
+            }
+        }
+        Ok(())
+    }
+
+    pub(crate) fn encode(&self, encoding: &CredentialValuesEncoding) -> Result<CredentialValues> {
+        match encoding {
+            CredentialValuesEncoding::Auto => {
+                let mut cred_values = MakeCredentialValues::default();
+                for (attribute, raw_value) in self.0.iter() {
+                    match raw_value {
+                        CredentialAttributeValue::Attribute(raw_value) => {
+                            cred_values.add_raw(attribute, raw_value)?
+                        }
+                        value => {
+                            return Err(err_msg!(
+                                "Encoding is not supported for credential value {:?}",
+                                value
+                            ));
+                        }
+                    }
+                }
+                Ok(cred_values.into())
+            }
+            encoding => Err(err_msg!(
+                "Credential values encoding {:?} is not supported",
+                encoding
+            )),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CredentialStatus {
+    #[serde(rename = "type")]
+    pub type_: CredentialStatusType,
+    pub id: RevocationRegistryDefinitionId,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum CredentialStatusType {
+    AnonCredsCredentialStatusList2023,
```

Comment:
Been a bit out of the loop, will this be a new revocation scheme or just a placeholder for future work?